src/data_structures/interval_tree.rs

```diff
@@ -272,13 +255,24 @@ fn swap_interval_data<N, D>(node_1: &mut Node<N, D>, node_2: &mut Node<N, D>) {
 }
 
 fn intersect<N: Ord>(range_1: &Range<N>, range_2: &Range<N>) -> bool {
-    range_1.end > range_2.start && range_1.start < range_2.end
+    range_1.start < range_1.end && range_2.start < range_2.end &&
```

Comment:
Are the additional checks necessary here? We already have restricted our ranges above, right?